src/lib.rs

```diff
@@ -439,6 +581,47 @@ impl Navigator {
         }
     }
 
+    /// Reads all sensors and stores on a single structure.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use navigator_rs::{Navigator, SensorData};
+    /// use std::thread::sleep;
+    /// use std::time::Duration;
+    ///
+    /// let mut nav = Navigator::new();
+    /// nav.init();
+    ///
+    /// loop {
+    ///     println!("---------------reading---------------");
+    ///     let sensor_data: SensorData = nav.read_all();
+    ///     println!("{:#?}", sensor_data);
+    ///     sleep(Duration::from_millis(1000));
+    /// }
+    /// ```
+    ///
+    /// ```no_run
+    /// use navigator_rs::{Navigator, SensorData};
+    /// use std::thread::sleep;
+    /// use std::time::Duration;
+    ///
+    /// let mut nav = Navigator::new();
+    /// nav.init();
+    ///
+    /// loop {
+    ///     let sensor_data: SensorData = nav.read_all();
+    ///     println!("---------------Reading ADC Channels---------------");
```

Comment:
This does not follow the style of the others headers, I would recommend to just remove all of them.

Reply:
removed.